Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm

```diff
@@ -277,9 +278,18 @@
         return;
     }
 
+    // Avoid running any more tasks before the runloop observer fires.
+    WebCore::WindowEventLoop::breakToAllowRenderingUpdate();
+
     startRenderingUpdateTimer();
 }
 
+void RemoteLayerTreeDrawingArea::updateRenderingTimerFired()
+{
+    updateRendering();
+    WebCore::WindowEventLoop::didCompleteRenderingUpdate();
```

Comment:
I guess an alternative design is to perform microtask check point after executing "update the rendering" step.